apps/prairielearn/src/lib/features/manager.ts

```diff
@@ -105,22 +101,27 @@ export class FeatureManager<FeatureName extends string> {
         ...DEFAULT_CONTEXT,
         ...context,
       },
-      IsFeatureEnabledSchema,
+      z.array(z.boolean()),
     );
 
-    if (featureIsEnabled.has_feature_grant) return true;
+    // The feature states are sorted in order of increasing specificity. If
+    // the last item is enabled, then the feature is enabled.
+    if (featureIsEnabled.at(-1)) return true;
```

Comment:
We could probably get away with doing this in SQL with `ORDER BY` and `LIMIT` to return this single boolean.

Reply:
I made this change in 57de206.